### PR TITLE
feat: definition and search updates

### PR DESCRIPTION
**Pull request by codemonkey800**
#1267

- Refactors the table header into a separate component
- Adds new table definition header component for increasing visibility of the dataset / deposition description 

### Browse Datasets

https://dev-table-def.cryoet.dev.si.czi.technology/browse-data/datasets

image

### Browse Depositions

https://dev-table-def.cryoet.dev.si.czi.technology/browse-data/depositions

image

### CODE REVIEW

**Comment by kev-zunshiwang** [comment]
Looking great! 

One nitpick: can you update the search bar right side padding to 24px so it aligns with the end of the table?

File: frontend/packages/data-portal/app/components/TablePageLayout/TablePageLayout.tsx [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 import { MAX_PER_PAGE } from 'app/constants/pagination'
 import { QueryParams } from 'app/constants/query'
 import { TestIds } from 'app/constants/testIds'
-import { LayoutContext, LayoutContextValue } from 'app/context/Layout.context'
-import { useI18n } from 'app/hooks/useI18n'
 import { cns } from 'app/utils/cns'
 
-import { ErrorBoundary } from './ErrorBoundary'
-import { TableCount } from './Table/TableCount'
+import { ErrorBoundary } from '../ErrorBoundary'
```
Comment (kne42) [inline frontend/packages/data-portal/app/components/TablePageLayout/TablePageLayout.tsx:12]: nit:
```suggestion
import { ErrorBoundary } from 'app/components/ErrorBoundary'
```

**Review by kne42 — APPROVED** [review]
lgtm!

File: frontend/packages/data-portal/app/components/Table/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
+export * from '../TablePageLayout/TableCount'
```
Comment (ehoops-cz) [inline frontend/packages/data-portal/app/components/Table/index.ts:1]: Not sure if this is the time for refactoring, but I recently learned about barrel files hurting performance:
https://marvinh.dev/blog/speeding-up-javascript-ecosystem-part-7/

**Review by ehoops-cz — APPROVED** [review]

File: frontend/packages/data-portal/app/components/Table/index.ts (hunk 2 above)
Comment (kne42) [reply frontend/packages/data-portal/app/components/Table/index.ts:1]: interesting. i wonder if this is the reason tooling like ESLint feels so slow lately

**Pull request merged**